File: backend/src/main/java/com/github/ryand6/sudokuweb/dto/entity/user/UserSettingsDto.java
package com.github.ryand6.sudokuweb.dto.entity.user;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.github.ryand6.sudokuweb.enums.Theme;
import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserSettingsDto {

    private Theme theme;

    private boolean opponentHighlightedSquaresEnabled;

    private boolean highlightedHousesEnabled;

    private boolean highlightedFirstsEnabled;

    private boolean audioEnabled;

    private boolean gameChatNotificationsEnabled;

    private boolean scoreNotificationsEnabled;

    private boolean streakNotificationsEnabled;

}

File: frontend/src/types/dto/entity/user/UserSettingsDto.ts
import type { Theme } from "@/types/enum/Theme"

export type UserSettingsDto = {
    theme: Theme,
    opponentHighlightedSquaresEnabled: boolean,
    highlightedHousesEnabled: boolean,
    highlightedFirstsEnabled: boolean,
    audioEnabled: boolean,
    gameChatNotificationsEnabled: boolean,
    scoreNotificationsEnabled: boolean,
    streakNotificationsEnabled: boolean
}